lib/sidekiq_unique_jobs/server/middleware.rb

```diff
@@ -7,45 +7,86 @@ module Server
     class Middleware
       include Extensions
 
-      attr_reader :unlock_order, :redis_pool
+      attr_reader :unlock_order,
+                  :redis_pool,
+                  :reschedule_on_lock_fail,
+                  :run_lock_retries,
+                  :run_lock_retry_interval
 
-      def call(worker, item, _queue, redis_pool = nil)
-        operative = true
+      def call(worker, item, _queue, redis_pool = nil, &blk)
         @redis_pool = redis_pool
-        decide_unlock_order(worker.class)
+        setup_options(worker.class)
+        send("#{unlock_order}_call", item, &blk)
+      end
+
+      # rubocop:disable MethodLength
+      def run_lock_call(item)
         lock_key = payload_hash(item)
-        unlock(lock_key, item) if before_yield?
+        run_lock = try_acquire_run_lock(lock_key, item)
+        if run_lock
```

Comment:
I prefer if we could get rid of all the nested ifs. Lets try and do early returns to avoid nesting.